.github/workflows/docker.yml

```diff
@@ -0,0 +1,68 @@
+name: Docker Build & Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      branch:
+        description: skynetlabs/skyd source branch (manual dispatch does NOT push to dockerhub)
+        type: string
+        required: true
+  pull_request:
+  push:
+    tags:
+      - v*
+
+jobs:
+  dockerfile-lint:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true # fail fast to prevent executing build-and-push
+      matrix:
+        flavor:
+          - bullseye-slim
+          - scratch
+    steps:
+      - uses: actions/checkout@v3
+      - uses: hadolint/hadolint-action@v2.1.0
+        with:
+          dockerfile: ${{ matrix.flavor }}/Dockerfile
+  build-and-push:
+    needs: [dockerfile-lint]
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        flavor:
+          - bullseye-slim
+          - scratch
+    steps:
+      - uses: actions/checkout@v3
+      - uses: docker/metadata-action@v4
+        id: meta
+        with:
+          images: skynetlabs/skyd
+          flavor: |
+            latest=${{ matrix.flavor == 'scratch' && 'auto' || 'false' }}
+            suffix=-${{ matrix.flavor }}
```

Comment:
> We're not producing tags 1.5.5, 1.5 and 1 yet deliberately because I was actually thinking whether scratch should be a default container or debian but I think scratch is safer and should be used unless the developer has particular needs and then different image should be selected.

updated in https://github.com/SkynetLabs/docker-skyd/pull/1/commits/b252eb707a6e344d7d59ec7582fb15e0c8c654ba to produce those from scratch